clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -32,7 +31,7 @@ void SYCLInstallationDetector::addSYCLIncludeArgs(
 
 // Unsupported options for SYCL device compilation.
 static ArrayRef<OptSpecifier> getUnsupportedOpts() {
-  return {
+  static std::vector<OptSpecifier> UnsupportedOpts = {
```

Comment:
Can't this function be `constexpr` instead of using a static variable?

Reply:
Declaring the function `constexpr` wouldn't address the lifetime issue that is fixed by this change. The function is called at run-time, so the data backing the array reference needs to persist until run-time.

Reply:
and C array: UnsupportedOpts[] = {
}

Reply:
Hmm, I was thinking that you'd end up w/ a ref to the constant in RO data, but maybe that's naive of me. Probably @vitalybuka's suggestion is more appropriate, then.